spec/models/manageiq/providers/hawkular/middleware_manager_spec.rb

```diff
@@ -6,4 +6,49 @@
   it ".description" do
     expect(described_class.description).to eq('Hawkular')
   end
+
+  describe "miq_id_prefix" do
+    let(:random_id) { SecureRandom.hex(10) }
+    let(:my_region) do
+      MiqRegion.my_region || FactoryGirl.create(:miq_region, :region => MiqRegion.my_region_number)
+    end
+    let(:random_region) do
+      region = Random.rand(1..99) while !region || region == my_region.region
```

Comment:
In this context, I'm testing `miq_id_prefix` which for any two instances of `MiddlewareManager` should comply with:
* It must return the same prefix if they live in the same region and they are pointing to the same provider (i.e. they come from the same database record, or they are two records from two different databases but are logically the same record).
* It must return different prefixes if they live in different regions, regardless if they point to the same provider.
* It must return different prefixes if they live in the same region, but they point to different providers.

These are the requirements of the generated prefix. But there is also the requirement that the prefix should be generated only from the data of one provider (you shouldn't pass the other provider as an argument) and the method should return something that must follow those rules.

I only need a `MiqRegion` that it's different to what `MiqRegion.my_region` returns. I don't know what `MiqRegion.my_region` will return, so I just randomly chose something else.

I'm not sure how this can lead to false positives since it is enough to have two different regions to have the tests do its work. In a broader view, I think the tests are kind of deterministic.